### DISCUSSION TITLE
fr/hub/inference-api/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#8756] [Docs]
# fr/hub/inference-api/

Accède aux capacités de détection d'objets de YOLOv8 via notre API RESTful. Apprends à utiliser l'API d'inférence YOLO avec Python ou cURL pour la détection rapide d'objets.

https://docs.ultralytics.com/fr/hub/inference-api/

**Comment by Hamzaboughfir123** [comment]
how can I retrieve API_KEY and model_ID

**Reply by glenn-jocher** [reply]
Hey there! 😊 To get your `API_KEY` and `model_ID`, you'll need to sign up or log in to the Ultralytics HUB. Once logged in, you can create a new model or use an existing one to get the `model_ID`. For the `API_KEY`, it's usually found in your account settings or dashboard.

Here's a quick Python example using these values:

```python
import requests

# Replace 'YOUR_API_KEY' and 'YOUR_MODEL_ID' with your actual API key and model ID
API_KEY = 'YOUR_API_KEY'
MODEL_ID = 'YOUR_MODEL_ID'
url = f"https://api.ultralytics.com/v1/predict/{MODEL_ID}"

headers = {"x-api-key": API_KEY}
data = {"size": 640, "confidence": 0.25, "iou": 0.45}

with open("path/to/your/image.jpg", "rb") as image_file:
    files = {"image": image_file}
    response = requests.post(url, headers=headers, files=files, data=data)

print(response.json())
```

Make sure to replace `'YOUR_API_KEY'` and `'YOUR_MODEL_ID'` with your actual API key and model ID. If you have any more questions or need further assistance, feel free to ask! 🚀
